### PR TITLE
ApplicationDebugger/array-transform: add AoT compilation options

### PR DESCRIPTION
**Pull request by barisaktemur**
# Description

Add AoT compilation options to CMake. Also includes test configurations.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

### CODE REVIEW

**Review by JoeOster — APPROVED** [review]

**Comment by tomlenth** [comment]
Nice to have would be a simple statement why AOT is helpful.  Maybe something like "AoT Compilation is particularly helpful in larger applications where compiling with debug information takes considerably longer."

File: Tools/ApplicationDebugger/array-transform/README.md [hunk 1]
```diff
@@ -133,6 +133,29 @@ Perform the following steps:
     $ make clean
     ```
 
+By default, CMake configures the build for Just-in-Time (JIT)
+compilation of the kernel.  However, it also offers an option for
+*Ahead-of-Time* (AoT) compilation.  To compile the kernel
+ahead-of-time for a specific device, set the `DPCPP_COMPILE_TARGET`
+option to the desired device during configuration.  For CPU, use the
+`cpu` value; for FPGA-emu, use the `fpga-emu` value.  Other values are
```
Comment (akertesz) [inline Tools/ApplicationDebugger/array-transform/README.md:141]: To the best of my knowledge, FPGA *only* supports AoT compilation -- if you are discussing AoT, I think you need to mention this.

Nevermind, I was mistaken. FPGA emulator supports both JiT and AoT, although in practice it's extremely rare to use JiT. I withdraw my concern.

**Review by akertesz — APPROVED** [review]

**Comment by barisaktemur** [comment]
> Nice to have would be a simple statement why AOT is helpful. Maybe something like "AoT Compilation is particularly helpful in larger applications where compiling with debug information takes considerably longer."

@tomlenth, I pushed a commit that adds this note.

**Comment by tomlenth** [comment]
looks great, thx.

**Comment by tomlenth** [comment]
closed in error... re-opening.

**Review by JoeOster — APPROVED** [review]

**Review by tomlenth — APPROVED** [review]

**Pull request merged**